### PR TITLE
Make basepersistence methods abstractmethod

### LINKED ISSUE #2231: Make BP.flush an abstract method
Don't know why we skipped that in #1905, but as `BasePersistence` is a interface, it makes sense for all methods to be abcs. As `flush` currently doesn't raise `NotImplementedError`, this is strictly speaking breaking, so putting this on the v14 milestone.

### PR DESCRIPTION
**Pull request by starry69**
### Checklist for PRs

- [x] Added `.. versionadded:: version`, `.. versionchanged:: version` or `.. deprecated:: version` to the docstrings for user facing changes (for methods/class descriptions, arguments and attributes)
- [x] Created new or adapted existing unit tests

Closes #2231

### CODE REVIEW

**Review by Bibo-Joshi — COMMENTED** [review]
Just one question: Is it worth adding `.. versionchanged::` directives stating that the methods are abstract since v14?

**Comment by starry69** [comment]
> Just one question: Is it worth adding `.. versionchanged::` directives stating that the methods are abstract since v14?

I think yes

**Comment by Poolitzer** [comment]
Isnt this supposed to merge to the V14 branch?

File: telegram/ext/basepersistence.py [hunk 1]
```diff
@@ -446,6 +446,8 @@ def get_callback_data(self) -> Optional[CDCData]:
 
         .. versionadded:: 13.6
 
+        .. versionchanged:: 14.0
```
Comment (Bibo-Joshi) [inline telegram/ext/basepersistence.py:449]: mh, this now just states that __something__ changed, but not what …

Comment (starry69) [reply telegram/ext/basepersistence.py:449]: Ohh, sorry i didn't knew much about sphinx doc directives..

**Pull request merged**